backend/typescript/services/implementations/camperService.ts

```diff
@@ -0,0 +1,96 @@
+import ICamperService from "../interfaces/camperService";
+import MgCamper, { Camper } from "../../models/camper.model";
+import MgCamp, { Camp } from "../../models/camp.model";
+import { CreateCamperDTO, CamperDTO } from "../../types";
+import { getErrorMessage } from "../../utilities/errorUtils";
+import logger from "../../utilities/logger";
+
+const Logger = logger(__filename);
+
+class CamperService implements ICamperService {
+  /* eslint-disable class-methods-use-this */
+  async createCamper(camper: CreateCamperDTO): Promise<CamperDTO> {
+    let newCamper: Camper;
+    let existingCamp: Camp | null;
+    try {
+      newCamper = await MgCamper.create({
+        firstName: camper.firstName,
+        lastName: camper.lastName,
+        age: camper.age,
+        parentName: camper.parentName,
+        contactEmail: camper.contactEmail,
+        contactNumber: camper.contactNumber,
+        camp: camper.camp,
+        hasCamera: camper.hasCamera,
+        hasLaptop: camper.hasLaptop,
+        allergies: camper.allergies,
+        additionalDetails: camper.additionalDetails,
+        dropOffType: camper.dropOffType,
+        registrationDate: camper.registrationDate,
+        hasPaid: camper.hasPaid,
+        chargeId: camper.chargeId,
+      });
+
+      try {
+        existingCamp = await MgCamp.findByIdAndUpdate(
+          camper.camp,
+          {
+            $push: { campers: newCamper.id },
```

Comment:
we'll also have to look into disallowing registration if the camp is at capacity, can create a new task for this though